File: apps/web/src/features/account/api/mappers.ts
/**
 * Data transformation functions for account API.
 */

import type { Database } from '@housing-platform/types';

import type { Profile, ProfileUpdateInput } from '../model';

export type ProfileRow = Profile;
export type ProfileUpdatePayload = Database['public']['Tables']['profiles']['Update'];

export function mapProfileRow(row: ProfileRow | null): Profile | null {
  return row;
}

export function toProfileUpdatePayload(input: ProfileUpdateInput): ProfileUpdatePayload {
  return {
    full_name: input.full_name,
    phone: input.phone,
    avatar_url: input.avatar_url,
    preferred_language: input.preferred_language,
    marketing_consent: input.marketing_consent,
  };
}

File: apps/backend-python/src/housing_platform/profile/mappers.py
from housing_platform.auth.models import UserRole
from housing_platform.db.models import Profiles
from housing_platform.profile.schemas import Profile


def map_profile(profile: Profiles) -> Profile:
    return Profile(
        id=profile.id,
        role=UserRole(profile.role),
        full_name=profile.full_name,
        phone=profile.phone,
        avatar_url=profile.avatar_url,
        preferred_language=profile.preferred_language,
        marketing_consent=profile.marketing_consent,
        created_at=profile.created_at,
        updated_at=profile.updated_at,
    )
